nixos/modules/services/web-apps/lasuite-docs.nix

```
@@ -0,0 +1,576 @@
+    mkPackageOption
+    mkOption
+    types
+    optional
+    optionalString
+    ;
+
+  cfg = config.services.lasuite-docs;
+
+  pythonEnvironment = mapAttrs (
+    _: value:
+    if value == null then
+      "None"
+    else if value == true then
+      "True"
+    else if value == false then
+      "False"
+    else
+      toString value
+  ) cfg.settings;
+
+  gunicornSettingsFile = pkgs.writeText "gunicorn-settings.py" ''
+    bind = [ "${cfg.bind}" ]
+    name = "impress"
+    python_path = "${cfg.backendPackage}/lib/docs/src/"
+
+    graceful_timeout = 90
+    timeout = 90
+    workers = 3
+
+    accesslog = "-"
+    errorlog = "-"
+    loglevel = "debug"
+  '';
+
+  commonServiceConfig = {
+    RuntimeDirectory = "lasuite-docs";
+    StateDirectory = "lasuite-docs";
+    WorkingDirectory = "/var/lib/lasuite-docs";
+
+    User = "lasuite-docs";
+    DynamicUser = true;
+    SupplementaryGroups = mkIf cfg.redis.createLocally [ "redis-lasuite-docs" ];
+    # hardening
+    AmbientCapabilities = "";
+    CapabilityBoundingSet = [ "" ];
+    DevicePolicy = "closed";
+    LockPersonality = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    PrivateTmp = true;
+    PrivateUsers = true;
+    ProcSubset = "pid";
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectProc = "invisible";
+    ProtectSystem = "strict";
+    RemoveIPC = true;
+    RestrictAddressFamilies = [
+      "AF_INET"
+      "AF_INET6"
+      "AF_UNIX"
+    ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    UMask = "0077";
+  };
+
+  preloadSecrets = ''
+    ${optionalString (
+      cfg.s3.accessKeyIDPath != null
+    ) "export AWS_S3_ACCESS_KEY_ID=$(cat ${cfg.s3.accessKeyIDPath})"}
+    ${optionalString (
+      cfg.s3.secretAccessKeyPath != null
+    ) "export AWS_S3_SECRET_ACCESS_KEY=$(cat ${cfg.s3.secretAccessKeyPath})"}
+    ${optionalString (
+      cfg.oidc.clientSecretPath != null
+    ) "export OIDC_RP_CLIENT_SECRET=$(cat ${cfg.oidc.clientSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.serverSecretPath != null
+    ) "export COLLABORATION_SERVER_SECRET=$(cat ${cfg.collaborationServer.serverSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.yproviderApiKeyPath != null
+    ) "export Y_PROVIDER_API_KEY=$(cat ${cfg.collaborationServer.yproviderApiKeyPath})"}
+  '';
+in
+{
+  options.services.lasuite-docs = {
+    enable = mkEnableOption "SuiteNumérique Docs";
+
+    backendPackage = mkPackageOption pkgs "lasuite-docs" { };
+
+    frontendPackage = mkPackageOption pkgs "lasuite-docs-frontend" { };
+
+    bind = mkOption {
+      type = types.str;
+      default = "127.0.0.1:8000";
+      description = ''
+        The server socket used to bind.
+      '';
```

```suggestion
      default = "unix:/run/lasuite-docs/gunicorn.sock";
      example = "127.0.0.1:8000";
      description = ''
        The path, host/port or file descriptior to bind the gunicorn socket to. 
        
        See  <https://docs.gunicorn.org/en/stable/settings.html#bind> for possible options.
      '';
```

For local deployments we should always prefer unix domain sockets.

From 11 years ago: https://nicisdigital.wordpress.com/2014/03/03/unix-domain-sockets-vs-loopback-tcp-sockets/